### PR TITLE
Feature: request

### PR DESCRIPTION
**Pull request by vadymmarkov**
1) Introduce `Requitable` protocol and `Message` struct.
2) Helper methods for `GET`, `POST`, `PUT`, etc.
3) ETags support.

### CODE REVIEW

File: Sources/Request/Message.swift [hunk 1]
```diff
@@ -0,0 +1,44 @@
+import Foundation
+
+public struct Message: Equatable {
```
Comment (vadymmarkov) [inline Sources/Request/Message.swift]: @hyperoslo/malibu Do you guys think `Message` is a good name here? I think it's better than `NetworkMessage`.

File: Sources/Networking.swift [hunk 2]
```diff
@@ -44,10 +53,49 @@ public class Networking {
...
+
+    ETagStorage().add(etag, forKey: key)
+  }
+}
+
+// MARK: - Requests
+
+extension Networking {
+
+  func get(request: Requestable) -> Promise<NetworkResult> {
```
Comment (vadymmarkov) [inline Sources/Networking.swift]: `get` or `GET`?

Comment (RamonGilabert) [inline Sources/Networking.swift]: I like `GET`.

Comment (zenangst) [inline Sources/Networking.swift]: I think I like `GET`

File: Sources/Request/Message.swift (hunk 1 above)
Comment (onmyway133) [inline Sources/Request/Message.swift]: I think everyone knows that it is related to Network

**Comment by zenangst** [comment]
:palm_tree:

**Comment by RamonGilabert** [comment]
Wanna see the next PR with the title: Change name.

**Pull request merged**